### PR TITLE
chore: Normalize project structure and README

### PR DESCRIPTION
**Pull request by asterwyx**
Rename types files according to the style of dtkcore. Delete some abundant include statements, and use forward declaration instead. Modify README.

### CODE REVIEW

**Comment by toberyan** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -11,18 +18,68 @@ Qt-based development template library.
...
+
+    ```bash
+    ./test-recoverage.sh
+    ```
+
+    to acquire test coverage info.
+2. Release config
+
+    ```bash
+    cmake -Bbuild -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr && cmake --build build
```
Comment (ComixHe) [inline README.md:64]: -DCMAKE_BUILD_TYPE=Release

File: README.zh_CN.md [hunk 2]
```diff
@@ -11,18 +16,68 @@
...
+
+    ```bash
+    ./test-recoverage.sh
+    ```
+
+    获取单元测试报告
+2. Release配置：
+
+    ```bash
+    cmake -Bbuild -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr && cmake --build build
```
Comment (ComixHe) [inline README.zh_CN.md:62]: 同上

**Review by Groveer — COMMENTED** [review]
/+1

**Review by dandan-y — COMMENTED** [review]
+1

**Review by xzl01 — APPROVED** [review]

**Pull request merged**